src/diffusers/pipelines/fabric/pipeline_fabric.py

```diff
@@ -0,0 +1,552 @@
+            generated image likely represents "not-safe-for-work" (nsfw) content, according to the `safety_checker`.
+
+        """
+
+        self.check_inputs(prompt, negative_prompt, liked, disliked)
+
+        device = self._execution_device
+        dtype = self.unet.dtype
+
+        latent_noise = torch.randn(num_images, 4, 64, 64, device=device, dtype=dtype)
+
+        positive_latents = (
+            self.preprocess_feedback_images(liked, self.vae, device, dtype)
+            if liked and len(liked) > 0
+            else torch.tensor([], device=device, dtype=dtype)
+        )
+        negative_latents = (
+            self.preprocess_feedback_images(disliked, self.vae, device, dtype)
+            if disliked and len(disliked) > 0
+            else torch.tensor([], device=device, dtype=dtype)
+        )
+
+        if isinstance(prompt, str) and prompt is not None:
+            batch_size = 1
+        elif isinstance(prompt, list) and prompt is not None:
+            batch_size = len(prompt)
+        else:
+            batch_size = None
+
+        prompt = [prompt] * num_images
+
+        if isinstance(negative_prompt, str):
+            negative_prompt = [negative_prompt] * num_images
+        elif isinstance(negative_prompt, list):
+            negative_prompt = negative_prompt
+        else:
+            assert len(negative_prompt) == num_images
+
+        (cond_prompt_embs, uncond_prompt_embs, null_prompt_emb) = self.initialize_prompts(
+            prompt + negative_prompt + [""], device
+        ).split([num_images, num_images, batch_size * num_images])
+
+        batched_prompt_embd = torch.cat([cond_prompt_embs, uncond_prompt_embs], dim=0)
+
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        latent_noise = latent_noise * self.scheduler.init_noise_sigma
+
+        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+
+        ref_start_idx = round(len(timesteps) * feedback_start_ratio)
+        ref_end_idx = round(len(timesteps) * feedback_end_ratio)
+
+        with self.progress_bar(total=num_inference_steps) as pbar:
+            for i, t in enumerate(timesteps):
+                sigma = self.scheduler.sigma_t[t] if hasattr(self.scheduler, "sigma_t") else 0
+                if hasattr(self.scheduler, "sigmas"):
+                    sigma = self.scheduler.sigmas[i]
+
+                alpha_hat = 1 / (sigma**2 + 1)
+
+                z_single = self.scheduler.scale_model_input(latent_noise, t)
+                z_all = torch.cat([z_single] * 2, dim=0)
+                z_ref = torch.cat([positive_latents, negative_latents], dim=0)
+
+                if i >= ref_start_idx and i <= ref_end_idx:
+                    weight_factor = max_weight
+                else:
+                    weight_factor = min_weight
+
+                pos_ws = (weight_factor, weight_factor * pos_bottleneck_scale)
```

Comment:
This is how neg_weights and pos_weights are calculated, It is exposed to the user in the form of max_weight, min_weight and neg_scale.